doc/api/stream.md

```diff
@@ -1725,6 +1725,52 @@ async function showBoth() {
 showBoth();
 ```
 
+### `readable.map(fn[, options])
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+* `fn` {Function|AsyncFunction} a function to map over every item in the stream.
+  * `data` {any} a chunk of data from the stream.
+  * `options` {Object}
+    * `signal` {AbortSignal} aborted if the stream is destroyed allowing to 
+      abort the `fn` call early.
+* `options` {Object}
+  * `concurrency` {number} the maximal concurrent invocation of `fn` to call 
+    on the stream at once. **Default:** `1`.
+  * `signal` {AbortSignal} allows destroying the stream if the signal is
+    aborted.
+* Returns: {Readable} a stream mapped with the function `fn`.
+
+This method allows mapping over the stream. The `fn` function will be called
+for every item in the stream. If the `fn` function returns a promise - that
+promise will be `await`ed before being passed to the result stream.
+
+Note that the order of results in the resulting stream is not guaranteed if
+`concurrency` is greater than `1`.
```

Comment:
@benjamingr This is not true.
```suggestion
```

Reply:
I noticed that this is the case currently - do we want to promise it?

Reply:
I guess if we ship it with this guarantee implicitly it won't matter since people will rely on it anyway, I'll remove